### PR TITLE
Binary search for characters

### PR DESCRIPTION
**Pull request by saur0n**
Implemented binary search of characters to avoid large sparce arrays of pointers to character definitions.
See #3.

### CODE REVIEW

File: src/fontrender.c [hunk 1]
```diff
@@ -15,8 +15,21 @@
 
 const struct glyph *font_get_glyph(const struct font *font, glyph_t glyph)
 {
-	if (glyph > font->max) return NULL;
-	return font->glyphs[glyph];
+	if (glyph > font->max)
+		return NULL;
+	
+	size_t first = 0, last = font->count;
```
Comment (chrisy) [inline src/fontrender.c:21]: `font->count` and `font->max` should be the same value now (perhaps offset by 1), yes? Is it worth consolidating?

File: src/fontem.c [hunk 2]
```diff
@@ -379,7 +379,7 @@ void store_glyph(FT_Face *face, FT_GlyphSlotRec *glyph,
 
 	// Append to the post
 	char *str = malloc(len + 100);
-	snprintf(str, len + 100, "\t[%u] = &%s,  /* '%s' */\n", ch, gname, mb(ch));
```
Comment (chrisy) [inline src/fontem.c:382]: Is it worth adding the hex value of `ch` inside the comment block? Purely for a diagnostic aid.

**Review by chrisy — COMMENTED** [review]
Good stuff. Two minor thoughts below.

File: src/fontrender.c (hunk 1 above)
Comment (saur0n) [reply src/fontrender.c:21]: No. For example, if font contains two characters, 'a' and 'z', `font->count` will be 2, but `font->max` will be 0x7a (code point of 'z').

Comment (chrisy) [reply src/fontrender.c:21]: Ah, so it still tracks the code point, not its index. Got it.

File: src/fontem.c (hunk 2 above)
Comment (saur0n) [reply src/fontem.c:382]: Done.

**Review by chrisy — APPROVED** [review]
Cool stuff

**Pull request merged**